hier_config/constructors.py

```diff
@@ -0,0 +1,331 @@
+from contextlib import suppress
+from itertools import islice
+from logging import getLogger
+from pathlib import Path
+from re import search, sub
+
+from hier_config.platforms.driver_base import HConfigDriverBase
+
+from .child import HConfigChild
+from .model import Dump, Platform
+from .platforms.arista_eos.driver import HConfigDriverAristaEOS
+from .platforms.arista_eos.view import HConfigViewAristaEOS
+from .platforms.cisco_ios.driver import HConfigDriverCiscoIOS
+from .platforms.cisco_ios.view import HConfigViewCiscoIOS
+from .platforms.cisco_nxos.driver import HConfigDriverCiscoNXOS
+from .platforms.cisco_nxos.view import HConfigViewCiscoNXOS
+from .platforms.cisco_xr.driver import HConfigDriverCiscoIOSXR
+from .platforms.cisco_xr.view import HConfigViewCiscoIOSXR
+from .platforms.generic.driver import HConfigDriverGeneric
+from .platforms.hp_comware5.driver import HConfigDriverHPComware5
+from .platforms.hp_procurve.driver import HConfigDriverHPProcurve
+from .platforms.hp_procurve.view import HConfigViewHPProcurve
+from .platforms.juniper_junos.driver import HConfigDriverJuniperJUNOS
+from .platforms.view_base import HConfigViewBase
+from .platforms.vyos.driver import HConfigDriverVYOS
+from .root import HConfig
+
+logger = getLogger(__name__)
+
+
+def get_hconfig_driver(platform: Platform) -> HConfigDriverBase:  # noqa: PLR0911
+    """Create base options on an OS level."""
+    match platform:
+        case Platform.ARISTA_EOS:
+            return HConfigDriverAristaEOS()
+        case Platform.CISCO_IOS:
+            return HConfigDriverCiscoIOS()
+        case Platform.CISCO_NXOS:
+            return HConfigDriverCiscoNXOS()
+        case Platform.CISCO_XR:
+            return HConfigDriverCiscoIOSXR()
+        case Platform.GENERIC:
+            return HConfigDriverGeneric()
+        case Platform.HP_PROCURVE:
+            return HConfigDriverHPProcurve()
+        case Platform.HP_COMWARE5:
+            return HConfigDriverHPComware5()
+        case Platform.JUNIPER_JUNOS:
+            return HConfigDriverJuniperJUNOS()
+        case Platform.VYOS:
+            return HConfigDriverVYOS()
+
+
+def get_hconfig_view(config: HConfig) -> HConfigViewBase:
+    match config.driver.platform:
```

Comment:
Add docstring describing that for custom Views and Drivers, the user will need to make a function like this one locally.